Source/WebKit/UIProcess/API/Swift/WebPage.swift

```diff
@@ -120,20 +125,125 @@ public class WebPage_v0 {
             createObservation(for: \.isLoading, backedBy: \.isLoading),
             createObservation(for: \.serverTrust, backedBy: \.serverTrust),
             createObservation(for: \.hasOnlySecureContent, backedBy: \.hasOnlySecureContent),
+            createObservation(for: \.isWritingToolsActive, backedBy: \.isWritingToolsActive),
             createObservation(for: \.themeColor, backedBy: \.themeColor),
         ]
     }
 
+    public static func handlesURLScheme(_ scheme: String) -> Bool {
+        WKWebView.handlesURLScheme(scheme)
+    }
+
     @discardableResult
     public func load(_ request: URLRequest) -> NavigationID? {
         backingWebView.load(request).map(NavigationID.init(_:))
     }
 
+    @discardableResult
+    public func load(_ data: Data, mimeType: String, characterEncoding: String.Encoding, baseURL: URL) -> NavigationID? {
+        let cfEncoding = CFStringConvertNSStringEncodingToEncoding(characterEncoding.rawValue)
+        guard cfEncoding != kCFStringEncodingInvalidId else {
+            preconditionFailure("\(characterEncoding) is not a valid character encoding")
+        }
+
+        guard let convertedEncoding = CFStringConvertEncodingToIANACharSetName(cfEncoding) as? String else {
+            preconditionFailure("\(characterEncoding) is not a valid character encoding")
+        }
+
+        return backingWebView.load(data, mimeType: mimeType, characterEncodingName: convertedEncoding, baseURL: baseURL).map(NavigationID.init(_:))
+    }
+
     @discardableResult
     public func load(htmlString: String, baseURL: URL) -> NavigationID? {
         backingWebView.loadHTMLString(htmlString, baseURL: baseURL).map(NavigationID.init(_:))
     }
 
+    @discardableResult
+    public func load(_ request: URLRequest, allowingReadAccessTo readAccessURL: URL) -> NavigationID? {
+        // `WKWebView` annotates this method as returning non-nil, but it may return nil.
```

Comment:
Also for later, but this seems worth fixing more generally for Swift, even if it has source breaking implications.

Reply:
Would this also lead to issues like this? https://github.com/WebKit/WebKit/pull/37519

Reply:
nope, since it's not async